### PR TITLE
fix #3323

### LINKED ISSUE #3323: DX-Site-Root: ZMI Nav-Tree is no longer expandable
After migrating to dx-site-root, the navtree within the zmi is no longer expandable

Screenshot 2021-09-20 at 09 10 52

https://github.com/plone/Products.CMFPlone/issues/2454 @jaroel @ale-rt

### PR DESCRIPTION
**Pull request by jensens**
see #3323

### CODE REVIEW

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/Portal.py [hunk 1]
```diff
@@ -139,6 +139,24 @@ def __before_publishing_traverse__(self, arg1, arg2=None):
...
+        obj_ids.sort()
+        getOb = self._getOb
+        for id in obj_ids:
+            obj = getOb(id)
+            if (
+                hasattr(aq_base(obj), 'isPrincipiaFolderish')
+                and obj.isPrincipiaFolderish
+            ):
+                result.append(obj)
+        return result
```
Comment (ale-rt) [inline Products/CMFPlone/Portal.py:158]: Untested, but it seems to me that something like this is preferable:
```
return sorted(
    (obj for obj in self.objectValues() if getattr(aq_base(obj), 'isPrincipiaFolderish', False),
    key=lambda obj: obj.getId(),
)
```
1. you do not append to a list
2. you save an hasattr
3. you do not fetch the objects from the DB one at a time

What do you think?

**Review by ale-rt — APPROVED** [review]
LGTM
I added a comment to suggest what I think it is a performance improvement but this is good to go!
Feel free to merge!

File: Products/CMFPlone/Portal.py (hunk 1 above)
Comment (jensens) [reply Products/CMFPlone/Portal.py:158]: That looks good. You may want to suggest a PR for OFS.ObjetcManager as well!

OT: you now there's a nice Github suggestion feature where I just would need to click on accept ;) ? Lazy me.

**Comment by jensens** [comment]
@jenkins-plone-org please run jobs

File: Products/CMFPlone/Portal.py [hunk 2]
```diff
@@ -137,6 +137,16 @@ def __before_publishing_traverse__(self, arg1, arg2=None):
 
         super(PloneSite, self).__before_publishing_traverse__(arg1, arg2)
 
+    # Concept from OFS.OrderSupport
+    @security.protected(permissions.AccessContentsInformation)
+    def tpValues(self):
+        # Return a list of subobjects, used by ZMI tree tag (and only there).
+        # see also https://github.com/plone/Products.CMFPlone/issues/3323
+        return sorted(
+            (obj for obj in self.objectValues() if getattr(aq_base(obj), 'isPrincipiaFolderish', False)),
```
Comment (ale-rt) [inline Products/CMFPlone/Portal.py:146]: Let me try:
```suggestion
            (
                obj
                for obj in self.objectValues()
                if getattr(aq_base(obj), "isPrincipiaFolderish", False)
            ),
```

**Review by ale-rt — APPROVED** [review]
Approved, with a suggestion (I hope I did it right)

File: Products/CMFPlone/Portal.py (hunk 2 above)
Comment (jensens) [reply Products/CMFPlone/Portal.py:146]: ah, and now I've been to fast.

Comment (jensens) [reply Products/CMFPlone/Portal.py:146]: we should run black over the code base

Comment (jensens) [reply Products/CMFPlone/Portal.py:146]: but not unless the big PR's are merged

**Pull request merged**